client/src/components/Workflow/Editor/Node.vue

```diff
@@ -300,60 +301,40 @@ export default {
                 };
             });
             this.initData(data);
-            Vue.nextTick(() => {
-                this.updateData(data);
-                this.$emit("onActivate", this);
-            });
+
+            // emit change completion event
+            this.showLoading = false;
+            this.$emit("onChange");
+            this.$emit("onActivate", this);
         },
         setAnnotation(annotation) {
-            if (this.annotationTimeout) {
-                clearTimeout(this.annotationTimeout);
-            }
-            this.annotationTimeout = setTimeout(() => {
-                if (annotation !== this.annotation) {
-                    this.annotation = annotation;
-                    this.$emit("onChange");
-                }
-            }, 100);
+            this.annotation = annotation;
+            this.$emit("onChange");
         },
         setLabel(label) {
-            if (this.labelTimeout) {
-                clearTimeout(this.labelTimeout);
-            }
-            this.labelTimeout = setTimeout(() => {
-                if (label !== this.label) {
-                    this.label = label;
-                    this.$emit("onChange");
-                }
-            }, 100);
+            this.label = label;
+            this.$emit("onChange");
         },
         setData(data) {
             this.config_form = data.config_form;
-            this.content_id = data.config_form?.id || data.content_id;
```

Comment:
@guerler I think we shouldn't have moved this to initData, that is the likely culprit for https://github.com/galaxyproject/galaxy/issues/13153, which is still happening when I backport https://github.com/galaxyproject/galaxy/pull/13185